src/System.CodeDom/tests/System/CodeDom/Compiler/CodeDomProviderTests.cs

```diff
@@ -134,6 +136,15 @@ public void CreateProvider_NoSuchLanguage_ThrowsConfigurationErrorsException(str
             AssertIsConfigurationErrorsException(ex2);
         }
 
+        [Fact]
+        public void CreateProvider_SerializeConfigurationErrorsException_ThrowsPlatformNotSupportedException()
+        {
+            Exception ex = Assert.ThrowsAny<Exception>(() => CodeDomProvider.CreateProvider(""));
+            MethodInfo method = typeof(BinaryFormatterHelpers).GetMethod(nameof(BinaryFormatterHelpers.AssertExceptionDeserializationFails));
+            MethodInfo genericMethod = method.MakeGenericMethod(new Type[] { ex.GetType() });
+            genericMethod.Invoke(ex, new object[0]);
```

Comment:
The test name says ThrowsPlatformNotSupportedException, but not sure how it is testing that

Reply:
Updated with this comment.